var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -79,6 +82,13 @@ class Cuda(Package):
 
     depends_on('libxml2', when='@10.1.243:')
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('--version', output=str)
```

Comment:
Could you please add:
```suggestion
        output = Executable(exe)('--version', output=str, error=str)
```
to avoid spurious output from failed detections? I'll try to put together a convenience function asap to hide this boilerplate, since these three lines are used quite frequently in all the PRs I have seen to detect external packages.